app/models/patient.rb

```diff
@@ -100,6 +100,31 @@ def self.trim_urgent_patients
     end
   end
 
+  def self.contacted_since(datetime)
+    # 1) filters to all patients with a call logged as Reached patient >= the date and counts them
+    # 2) identifies how many of those patients have been marked pledge_sent this week
+
+    # returns hash
+    # {
+    #   since: datetime,
+    #   contacts: (count of patients reached)
+    #   first_contacts: (count of first time reached patients),
+    #   pledges_sent: (count of pledges sent)
+    # }
+
+
+    Patient.each do |patient|
```

Comment:
Swap this out with `all.each do |patient|` -- the `Patient` part of this is inferred from it being a class method.